lib/active_model/serializer/adapter/json.rb

```diff
@@ -4,44 +4,130 @@ class ActiveModel::Serializer::Adapter::Json < ActiveModel::Serializer::Adapter
 
         def serializable_hash(options = nil)
           options ||= {}
-          if serializer.respond_to?(:each)
-            result = serializer.map { |s| FlattenJson.new(s).serializable_hash(options) }
-          else
-            hash = {}
 
-            core = cache_check(serializer) do
-              serializer.attributes(options)
+          hash =
+            if serializer.respond_to?(:each)
+              serializable_hash_for_collection(options)
+            else
+              serializable_hash_for_single_resource
```

Comment:
Two things then:
+ either there are genuinely no options, and we might as well get rid of it for now (just keeping an underscored parameter for the constructor), or
+ there actually are some options, and we better not remove them all, otherwise people will be angry :D